index.js

```diff
@@ -16,19 +34,18 @@ module.exports = function() {
         if (matched) {
           let id = matched[1]
           if (!!id.length) {
-            if (!node.data) {
-              node.data = {}
-            }
-            if (!node.data.hProperties) {
-              node.data.hProperties = {}
-            }
             node.data.id = node.data.hProperties.id = id
 
             string = string.substring(0, matched.index)
             lastChild.value = string
+            return
           }
         }
       }
+
+      // If no custom id was found, use default instead
+      let fullText = extractText(node.children)
```

Comment:
Please add a new option for enable/disable default id, and the default value is disable the feature for forward compatibility.

Reply:
Ok, I have made the requested change, as well as moved some code around and added more tests.